### PR TITLE
Update `GIT_PUSH` policies

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Update `GIT_PUSH` policies
* Split `GIT_PUSH` policies into policies for proposed runs and tracked runs

## why
*  We used `affected_files := input.pull_request.diff` to get the list of affected files. This worked in almost all cases: 
   - Push to PR
   - Merge PR to the default branch
 
   But it did not work for tracked runs in these cases (for which we need to use `affected_files := input.push.affected_files`):
   - Manually (from GitHub UI) update a file (component or YAML) and push to the default branch
   - When a bot updates files and pushes to the default branch automatically without opening a PR
 
* Split `GIT_PUSH` policies into policies for proposed runs and tracked runs 
  - Policies for proposed runs and tracked runs are different now and we can add more differences in the future 
  - Easier to manage and update separately
  - The policies' file names now better reflect the purpose of the policies

### CODE REVIEW

File: catalog/policies/git_push.tracked-run.rego
Comment (nitrocode) [inline catalog/policies/git_push.tracked-run.rego:12]: Oh very nice! Thank you for adding `.json` to this.

File: variables.tf
Comment (nitrocode) [inline variables.tf:213]: nit: this is weird but super low priority.

File: catalog/policies/git_push.tracked-run.rego
Comment (nitrocode) [inline catalog/policies/git_push.tracked-run.rego:57]: if the name of the stack is `v1-datadog.yaml`, would this cause issues here ?

**Review by nitrocode — DISMISSED** [review]
Looks good to me. I'm still new to rego so it might be wise to get a second review.

**Review by Nuru — DISMISSED** [review]

File: catalog/policies/git_push.tracked-run.rego
Comment (aknysh) [reply catalog/policies/git_push.tracked-run.rego:57]: it will be caught by the main code that detects the `deps:` labels.
It's here for historical reasons, I did not remove it to not make too many changes at once.

File: variables.tf
Comment (aknysh) [reply variables.tf]: fixed

**Review by nitrocode — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Pull request merged**
